### PR TITLE
fix(rds): proxy for db cluster fails with model validation error

### LINKED ISSUE #8885: [DatabaseProxy] Model validation failed (#: required key [TargetGroupName] not found)
I tried DatabaseProxy with Database Cluster (Aurora, Postgres) but it's failed as titled.

### Reproduction Steps

```typescript
    const databaseUser = 'test';
    const secret = new secretsmanager.Secret(this, 'secret', {
      generateSecretString: {
        generateStringKey: 'password',
        secretStringTemplate: `{"username": "${databaseUser}"}`,
        excludePunctuation: true
      }
    });

    const cluster = new rds.DatabaseCluster(this, 'Database', {
      engine: rds.DatabaseClusterEngine.AURORA_POSTGRESQL,
      engineVersion: '11.7',
      defaultDatabaseName: 'mydatabase',
      masterUser: {
        username: databaseUser,
        password: secret.secretValueFromJson('password')
      },
      instanceProps: {
        instanceType: ec2.InstanceType.of(
          ec2.InstanceClass.R5,
          ec2.InstanceSize.LARGE
        ),
        vpcSubnets: {
          subnetType: ec2.SubnetType.PRIVATE
        },
        vpc
      },
      parameterGroup: new rds.ClusterParameterGroup(this, 'ParameterGroup', {
        family: 'aurora-postgresql11',
        parameters: {
          application_name: 'test',
        },
      }),
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const proxy = new rds.DatabaseProxy(this, 'DatabaseProxy', {
      dbProxyName: 'test-proxy',
      debugLogging: true,
      iamAuth: false,
      requireTLS: true,
      secret,
      vpc,
      vpcSubnets: {
        subnetType: ec2.SubnetType.PRIVATE,
      },
      proxyTarget: rds.ProxyTarget.fromCluster(cluster),
    });
```

### Error Log

```
Model validation failed (#: required key [TargetGroupName] not found)
```

### Environment

  - **CLI Version      :** 1.49.1
  - **Framework Version:** 1.49.1
  - **Node.js Version:** v13.7.0
  - **OS               :** 10.14.6
  - **Language (Version):** TypeScript (3.7.5)

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by civilizeddev**
fixes #8885
follows #8476

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/proxy.ts [hunk 1]
```diff
@@ -408,20 +408,30 @@ export class DatabaseProxy extends cdk.Resource
...
       dbProxyName: this.dbProxyName,
       dbInstanceIdentifiers,
-      dbClusterIdentifiers: bindResult.dbClusters?.map((c) => c.clusterIdentifier),
+      dbClusterIdentifiers,
       connectionPoolConfigurationInfo: toConnectionPoolConfigurationInfo(props),
     });
+
+    // Currently(2020-07-04), this property must be set to default.
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxytargetgroup.html#TargetGroupName-fn::getatt
+    proxyTargetGroup.addOverride('Properties.TargetGroupName', 'default');
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/proxy.ts:434]: Overrides are set for CloudFormation properties that are not yet supported in the CDK. This is a CloudFormation return value. 

Why is this being set?

File: packages/@aws-cdk/aws-rds/lib/proxy.ts [hunk 2]
```diff
@@ -408,20 +408,30 @@ export class DatabaseProxy extends cdk.Resource
     this.endpoint = this.resource.attrEndpoint;
 
     let dbInstanceIdentifiers: string[] | undefined;
-    if (bindResult.dbClusters) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/proxy.ts:411]: Please add a unit test for the case that was previously wrong and is fixed in this PR.

File: packages/@aws-cdk/aws-rds/lib/proxy.ts (hunk 1 above)
Comment (civilizeddev) [reply packages/@aws-cdk/aws-rds/lib/proxy.ts:434]: There seems to be a difference between [the CloudFormation resource document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxytargetgroup.html#TargetGroupName-fn::getatt) and what CloudFormation actually does.

Currently CloudFormation requires `TargetGroupName` in props of `ProxyTargetGroup`, but it is not in `CfnProxyTargetGroupProps`

There is no normal way to set `TargetGroupName`, but through `addOverride`.

image

The behavior seems to be changed around DB Proxy's GA.

Comment (nija-at) [reply packages/@aws-cdk/aws-rds/lib/proxy.ts:434]: Thanks for the explanation.

**Pull request merged**